pkg/kubelet/cm/dra/checkpoint/checkpointer.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package checkpoint
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager"
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager/errors"
+	state "k8s.io/kubernetes/pkg/kubelet/cm/dra/state/v1"
+)
+
+type Checkpointer interface {
+	GetOrCreate() (state.ClaimInfoStateList, error)
+	Store(state.ClaimInfoStateList) error
+}
+
+type checkpointer struct {
```

Comment:
this should be called `checkpointerV1`

Reply:
I thought it should always work with the latest API version.